src/ol/source/imagewmssource.js

```diff
@@ -246,7 +222,23 @@ ol.source.ImageWMS.prototype.getImage =
   extent[1] = centerY - imageResolution * height / 2;
   extent[3] = centerY + imageResolution * height / 2;
 
-  this.imageExtent_ = extent;
+  var image = this.image_;
+  if (!goog.isNull(image) &&
+      this.renderedRevision_ == this.getRevision() &&
+      image.getResolution() == resolution &&
+      image.getPixelRatio() == pixelRatio &&
+      ol.extent.containsExtent(image.getExtent(), extent)) {
```

Comment:
But isn't the original issue back with this? Shouldn't `this.image_.getExtent()` be used instead of `extent` here?